federationtypes.go

```diff
@@ -0,0 +1,97 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// A SendRequest is the content of a request to PUT /_matrix/federation/v1/send/{txnID}/
+type SendRequest Transaction
```

Comment:
I don't like this for 2 reasons:
 - We usually do this so we can have the compiler help us out. The one place where this is used takes an `interface{}` so there is no benefit to typing this.
 - Casting to this type make it look like it has side-effects:
    ```go
   r := SendRequest(txn)
    ```

I would stick with `Transaction` and remove this type.